modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/mdl/MaterialComponent.scala

```diff
@@ -0,0 +1,38 @@
+package edu.gemini.seqexec.web.client.mdl
+
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.all._
+import japgolly.scalajs.react.CompScope
+
+import scala.scalajs.js
+
+/**
+  * Support for MDL components
+  */
+object MaterialComponent {
+  type ScopeType = CompScope.DuringCallbackM[(ReactTag, Boolean), Unit, Unit, TopNode]
+
+  val rc = ReactComponentB[(ReactTag, Boolean)]("MaterialComponent")
+    .render_P(props => props._1)
+    .componentDidMount(afterMount)
+    .build
+
+  def apply(props: (ReactTag, Boolean)): ReactComponentU[(ReactTag, Boolean), Unit, Unit, TopNode] = {
+    rc(props)
+  }
+
+  private def upgrade(scope: ScopeType): Callback = Callback {
+    js.Dynamic.global.componentHandler.upgradeElement(scope.getDOMNode())
+    if (scope.props._2) {
+      val children = scope.getDOMNode().children
+      (0 until children.length).foreach(i =>
+        js.Dynamic.global.componentHandler.upgradeElement(children(i))
+      )
+    }
+  }
```

Comment:
This code is unused as I'm not using MDL anymore but I wanted to keep it just in case
It essentially calls the MDL function `componentHandler.upgradeElement` on the current DOM node and its children 
I'll add some comments though